packages/block-editor/src/components/typewriter/index.js

```diff
@@ -97,44 +120,66 @@ class Typewriter extends Component {
 			return;
 		}
 
+		const editableNodes = this.ref.current.querySelectorAll( '[contenteditable="true"]' );
+		const lastEditableNode = editableNodes[ editableNodes.length - 1 ];
+		const isLastEditableNode = lastEditableNode === document.activeElement;
+
 		// The scroll container may be different depending on the viewport
 		// width.
+		// Nested condition: if the scroll position is at the start and the
+		// active editable element is the last one, do not scroll the page.
+		// The typewriter effect should not kick in until an empty page has been
+		// filled or the user scrolls intentionally down.
 		if ( scrollContainer === document.body ) {
+			if ( isLastEditableNode && window.scrollY === 0 ) {
+				return;
+			}
+
 			window.scrollBy( 0, diff );
 		} else {
+			if ( isLastEditableNode && scrollContainer.scrollTop === 0 ) {
+				return;
+			}
+
 			scrollContainer.scrollTop += diff;
 		}
 	}
 
-	onMouseDown() {
-		document.addEventListener( 'selectionchange', this.computeCaretRectOnSelectionChange );
-	}
-
-	onTouchStart() {
+	/**
+	 * Adds a `selectionchange` listener to reset the scroll position to be
+	 * maintained.
+	 */
+	addSelectionChangeListener() {
 		document.addEventListener( 'selectionchange', this.computeCaretRectOnSelectionChange );
 	}
 
-	onKeyDown() {
+	onKeyDown( event ) {
+		event.persist();
 		// Ensure the any remaining request is cancelled.
 		window.cancelAnimationFrame( this.rafId );
 		// Use an animation frame for a smooth result.
-		this.rafId = window.requestAnimationFrame( this.maintainCaretPosition );
-		// In rare cases, the selection is not updated before the animation
-		// frame. This selection change listener is a back up.
-		document.addEventListener( 'selectionchange', this.maintainCaretPositionOnSelectionChange );
+		this.rafId = window.requestAnimationFrame( () =>
+			this.maintainCaretPosition( event )
+		);
 	}
 
 	render() {
+		// There are some issues with Internet Explorer, which are probably not
+		// worth spending time on. Let's disable it.
+		if ( isIE ) {
+			return null;
```

Comment:
We need to return `this.props.children` here instead of just `null`, otherwise nothing will render inside the editor content area.

Reply:
Whoops. :)